udemy_exercises/songs/Album.java

```diff
@@ -1,7 +1,213 @@
+        return false;
+    }
+
+    public boolean addToPlaylist(String title, LinkedList<Song> playList) {
+        Song checkedSong = findSong(title);
+        if(checkedSong != null) {
+            playList.add(checkedSong);
+            return true;
+        }
+        System.out.println("The song " + title + " is not in this album");
+        return false;
+    }
+
+    public static void main(String[] args) {
+        Album album = new Album("DIVISIONS", "STARSET");
+        album.addSong("A BRIEF HISTORY OF THE FUTURE", 1.38);
+        album.addSong("MANIFEST", 4.27);
+        album.addSong("ECHO", 3.37);
+        album.addSong("WHERE THE SKIES END", 6.34);
+        album.addSong("PERFECT MACHINE", 5.24);
+        album.addSong("TELEKINETIC", 5.13);
+        album.addSong("STRATOSPHERE", 4.17);
+        album.addSong("FAULTLINE", 3.36);
+        album.addSong("SOLSTICE", 5.42);
+        album.addSong("TRIALS", 4.19);
+        album.addSong("WAKING UP", 3.50);
+        album.addSong("OTHER WORLDS THAN THESE", 4.18);
+        album.addSong("DIVING BELL", 5.38);
+        albums.add(album);
+
+        album = new Album("MNQN", "MNQN");
+        album.addSong("Config.Sys", 1.29);
+        album.addSong("iGhost", 3.59);
+        album.addSong("Amphetamine", 6.15);
+        album.addSong("Animal Oddity", 6.10);
+        album.addSong("Out Of Sight Out Of Mind", 6.15);
+        album.addSong("Man on the Moon", 3.51);
+        album.addSong("What Have You Become?", 4.22);
+        album.addSong("On My Mind", 4.37);
+        album.addSong("Don't Get It", 4.47);
+        album.addSong("Zombie", 5.43);
+        album.addSong("Noir", 4.34);
+        album.addSong("Invincible", 4.47);
+        album.addSong("Terminal", 5.24);
+        album.addSong("What Dreams My Come", 1.13);
+        albums.add(album);
+
+        LinkedList<Song> playList = new LinkedList<Song>();
+        albums.get(0).addToPlaylist("TRIALS", playList);
+        albums.get(0).addToPlaylist("ECHO", playList);
+        albums.get(0).addToPlaylist("Monster", playList); // shouldn't work
+        albums.get(0).addToPlaylist(2, playList);
+        albums.get(1).addToPlaylist(3, playList);
+        albums.get(1).addToPlaylist(6, playList);
+        albums.get(1).addToPlaylist(24, playList); // There is no track 24
+
+        play(playList);
+    }
+
+    private static void stop() {
+        System.out.println("Playlist complete.");
+    }
+
+    private static void play(LinkedList<Song> playList) {
+        Scanner scanner = new Scanner(System.in);
+        boolean quit = false;
+        boolean forward = true;
+        ListIterator<Song> listIterator = playList.listIterator();
+        if(playList.size() == 0) {
+            System.out.println("No songs in playlist");
+            return;
+        } else {
+            System.out.println("Now playing " + listIterator.next().toString());
+            printMenu();
+        }
+        while(!quit) {
```

Comment:
This logic seems to work, but very challenging to read.  Nested logic blocs (e.g. an 'if' within an 'if') and a list of logic blocks (e.g. an 'if' immediately followed by an 'if', or multiple 'else's) are usually to be avoided, and can replaceable with improved encapsulation.  Start by considering how you might encapsulate each action (the 'cases') into a separate method.